### PR TITLE
Activate filesource to list offline regions

### LINKED ISSUE #10470: OfflineManager.ListOfflineRegionsCallback doesn't work
**Platform:** Android 
**Mapbox SDK version:** 5.2.0-beta.3

I made a sample project which can reproduce a below bug.
https://github.com/rakuishi/mapbox-gl-native-issue

I think OfflineManager depends on MapView's lifecycle.
Because, after removing `mapView.onStop()`, callback works fine.

### Steps to trigger behavior

 1. Launch [MainActivity](https://github.com/rakuishi/mapbox-gl-native-issue/blob/master/app/src/main/java/com/rakuishi/mapbox_offlinemanager/MainActivity.java) which it has a MapView.
 2. Launch [Main2Activity](https://github.com/rakuishi/mapbox-gl-native-issue/blob/master/app/src/main/java/com/rakuishi/mapbox_offlinemanager/Main2Activity.java) from MainActivity.
 3. Then, call `OfflineManager.getInstance(context).listOfflineRegions()` by a touch event, but it's callback(`OfflineManager.ListOfflineRegionsCallback`) doesn't work.

### Expected behavior

- `OfflineManager.getInstance(context).listOfflineRegions()` works correctly.

### Actual behavior

- `OfflineManager.getInstance(context).listOfflineRegions()` doesn't work.

### PR DESCRIPTION
**Pull request by tobrun**
closes #10470

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java [hunk 1]
```diff
@@ -149,15 +149,16 @@ private Handler getHandler() {
...
     listOfflineRegions(fileSource, new ListOfflineRegionsCallback() {
 
       @Override
       public void onList(final OfflineRegion[] offlineRegions) {
         getHandler().post(new Runnable() {
           @Override
           public void run() {
+            fileSource.deactivate();
             callback.onList(offlineRegions);
           }
```
Comment (osana) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java]: Should fileSource.deactivate();
happen after the callback.onList(offlineRegions); ?

Comment (osana) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java]: I rebased osana-offline branch and was not able to see listed offline regions any more.
Reversing the order - fixed my issue

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java [hunk 2]
```diff
@@ -168,6 +169,7 @@ public void onError(final String error) {
         getHandler().post(new Runnable() {
           @Override
           public void run() {
+            fileSource.deactivate();
```
Comment (osana) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java]: Probably we should do the callback first and then deactivate the fileSource

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java (hunk 1 above)
Comment (tobrun) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java]: > I rebased osana-offline branch and was not able to see listed offline regions any more.
Reversing the order - fixed my issue

startled that this is the case, could you follow up in your branch with reversing?

Comment (osana) [reply platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/offline/OfflineManager.java]: I updated the branch with that change.

**Pull request merged**